libbeat/cmd/instance/beat.go

```diff
@@ -331,11 +333,17 @@ func (b *Beat) createBeater(bt beat.Creator) (beat.Beater, error) {
 		}
 	}
 
+	tracer, err := apm.NewTracer(b.Info.Beat, b.Info.Version)
```

Comment:
We should probably close the `apm.DefaultTracer` -- otherwise it will attempt to communicate with an APM Server on localhost.

Reply:
I think for APM Server we'll want to inject the server's custom tracer, so it can send events to itself without additional config.

I can see a couple of options:
 - add a `Tracer` field to the `Settings` struct, and use it if non-nil; otherwise create a new tracer
 - alternatively, just use `apm.DefaultTracer`

There's still going to be some awkwardness injecting the server's custom tracer: we would need to create a sort of "placeholder" tracer transport which we update once the server is active. That would be isolated to apm-server though.

Reply:
Sounds good. I'd rather leave that for a follow up, thou.